### ISSUE TITLE
Build error with build id #15980413

### ISSUE THREAD

**Issue by karcheba1** [readthedocs/readthedocs.org#8889]
## Details:

* Project URL: https://readthedocs.org/projects/vcfempy/
* Build URL(if applicable): https://readthedocs.org/projects/vcfempy/builds/15980413/
* Read the Docs username(if applicable): karcheba1-admin

## Expected Result

Building docs from public repo https://github.com/karcheba1/vcfempy.git
Expected html docs to be built? First time building docs on RTD, and have not succesfully built yet, so could be doing something wrong.

## Actual Result

Build apparently fails at the git clone stage. I am not really sure what is going wrong, as it is a public repo. I previously reported this issue with the repo hosted on Bitbucket. Trying to resolve it, I have since changed the repo host over to GitHub, and I am sure it is setup as a public repo. I setup the RTD project by importing the project from my connected GH account and there was a message confirming that the webhook was setup correctly.

I have test built the docs locally (in repo root ./docs/) using
sphinx-apidoc -f -e -l -o source/ ../src/vcfempy/
sphinx-build -M html source/ build/ -a
and this works correctly. I have also setup the Makefile in ./docs/ to execute these commands. The docs build without issue locally. Any help that you can provide is appreciated!

![Screenshot 2022-02-03 174555](https://user-images.githubusercontent.com/98863261/152453814-39e28fc3-7e3f-43cb-94b9-875532ab9618.png)

**Comment by karcheba1** [comment]
Hi Manuel,

Thanks for following up. Yes, I did work out what the problem was and able to find a solution, also after discussing with Anthony Johnson through the ***@***.******@***.***> email support.

The problem turned out to result from the fact that when I initialized the repo for this project, the ‘main’ branch included single quotes (unintentional on my part, something within git or the terminal I am using, which is a Ubuntu 20.04.3 terminal running under WSL on a Windows 11 machine). The problem is that the single quotes need to be escaped \’ to work properly because if you just ask for branch ‘main’ like “git checkout ‘main’” then git does not really see the single quotes as part of the name and interprets that the same as “git checkout main”. On my own machine, I found to switch back to the ‘main’ branch, I would have to use “git checkout \’main\’”, which was tedious but still worked. The RTD command to clone the repo I think targeted the ‘main’ branch specifically, but did not escape the single quotes, and ran into an issue because the repo did not have a main branch, only a ‘main’ branch. If, for some reason, I had a ‘main’ branch and a main branch, the command would likely work, but would perhaps not build docs for the branch I intended.

The solution in my case was to create a new main branch (no single quotes), set that as the default/latest branch of the repo, and delete the ‘main’ branch to avoid future confusion. Then I set the new main branch as the “latest” branch for build on RTD. It could probably be resolved on the RTD side also by ensuring that escape characters are included in the clone command if the name of the selected branch for the build includes characters that need to be escaped.

Cheers,
Brandon
